website/source/docs/guides/connect-gateways.md

```diff
@@ -0,0 +1,421 @@
+---
+layout: "docs"
+page_title: "Connecting Services Across Datacenters"
+sidebar_current: "docs-guides-connect-gateways"
+description: |-
+  Connect services and secure inter-service communication across datacenters
+  using Consul Connect and mesh gateways.
+---
+## Introduction
+
+Consul Connect is Consul’s service mesh offering, which allows users to observe
+and secure service-to-service communication. Because Connect implements mutual
+TLS between services, it also enabled us to build mesh gateways, which provide
+users with a way to help services in different datacenters communicate with each
+other. Mesh gateways take advantage of Server Name Indication (SNI), which is an
+extension to TLS that allows them to see the destination of inter-datacenter
+traffic without de-encrypting the message payload.
+
+Using gateways for inter-datacenter communication can prevent each Connect proxy
+from needing an accessible IP address, and frees operators from worrying about
+IP address overlap between datacenters.
+
+In this guide, you will configure Consul Connect across multiple Consul
+datacenters and use mesh gateways to enable inter-service traffic between them.
+
+Specifically, you will:
+
+1. Enable Connect in both datacenters
+1. Deploy the two mesh gateways
+1. Register services and Connect sidecar proxies
+1. Configure intentions
+1. Test that your services can communicate with each other
+
+For the remainder of this guide we will refer to mesh gateways as "gateways".
+Anywhere in this guide where you see the word gateway, assume it is specifically
+a mesh gateway (as opposed to an API or other type of gateway).
+
+## Prerequisites
+
+To complete this guide you will need two wide area network (WAN) joined Consul
+datacenters with access control list (ACL) replication enabled. If you are
+starting from scratch, follow these guides to set up your datacenters, or use
+them to check that you have the proper configuration:
+
+- [Deployment Guide](/consul/datacenter-deploy/deployment-guide)
+- [Securing Consul with ACLs](/consul/security-networking/production-acls)
+- [Basic Federation with WAN Gossip](/consul/security-networking/datacenters)
+
+You will also need to enable ACL replication, which you can do by following the [ACL Replication for Multiple Datacenters](/consul/day-2-operations/acl-replication) guide with the following modification.
+
+When creating the [replication token for ACL
+management](/consul/day-2-operations/acl-replication#create-the-replication-token-for-acl-management),
+it will need the following policy:
+
+```json
```

Comment:
The HCL form of the policy would look much nicer.

```hcl
acl = "write"
operator = "write"
service_prefix "" {
   policy = "read"
}
```

Also of note is that its a service_prefix rule not a service rule.

Reply:
I think @freddygv's suggestion was to stay consistent with the language for all our snippets, which is why I decided to convert it. Probably worth some thought about whether or not we care about being consistent. I think the ACL docs are one of the only places in the documentation where we use HCL, perhaps because the JSON gets extra ugly there.